c_emulator/riscv_sim.c

```diff
@@ -68,6 +68,7 @@ size_t spike_dtb_len = 0;
 char *sig_file = NULL;
 uint64_t mem_sig_start = 0;
 uint64_t mem_sig_end = 0;
+uint64_t signature_granularity = 4;
```

Comment:
Why the size of this is 64 bit, and then you are using int in the for loop ?

Reply:
Apart from that, I dont have anything else to comment on, and this looks and works fine to me.

Reply:
Oh actually, it might be helpfull to assert that signature_granularity is divisable by 4 !

Reply:
sure I'll change the signature_granularity size to int.
About "Δ mem_sig is divisable by signature_granularity" can you please elaborate it a little? like when Δ mem_sig is 
 not divisable by signature_granularity, is the signature granularity is invalid or Δ mem_sig.

Reply:
Maybe this is unnecessary, do we know for sure that the addresses are word-aligned or is there any alignment restriction on them? In the last iteration of the loop, if addr < mem_sig_end, then we will increment by signature_granularity, which could make our address outside the signature region?